File: Day 08/Stock_Buy_And_Sell_II.py
class Solution:
    def maximumProfit(self, prices):
        min_price = prices[0]
        max_profit = 0
        
        for i in range(1, len(prices)):
            if prices[i] < min_price:
                min_price = prices[i]
            
            profit = prices[i] - min_price
            if profit > max_profit:
                max_profit = profit
        
        return max_profit

if __name__ == "__main__":
    prices = [7, 1, 5, 3, 6, 4]
    sol = Solution()
    result = sol.maximumProfit(prices)
    print(f"Maximum Profit: {result}")

File: Day 08/Stock_Buy_And_Sell_II.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
    public:
      int maximumProfit(vector<int> &prices) {
          int minPrice = prices[0];
          int maxProfit = 0;
          for(int i = 1; i < prices.size(); i++){
              if(prices[i] < minPrice){
                  minPrice = prices[i];
              }
              int profit = prices[i] - minPrice;
              if(profit > maxProfit){
                  maxProfit = profit;
              }
          }
          
          return maxProfit;
      }
  };

int main() {
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    Solution sol;
    int result = sol.maximumProfit(prices);
    cout << "Maximum Profit: " << result << endl;
    return 0;
}